app/WebParsing/ParsecCombinators.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE OverloadedStrings #-}
+module WebParsing.ParsecCombinators
+    (getCourseFromTag,
+     findCourseFromTag,
+     getPostType,
+     extractPostType,
+     findPostType,
+     getDepartmentName,
+     isDepartmentName,
+     parsingAlgoOne) where
+
+import qualified Text.Parsec as P
+import Text.Parsec ((<|>))
+import qualified Data.Text as T
+import Data.Functor.Identity
+
+getCourseFromTag courseTag = do
+    let course = P.parse findCourseFromTag "(source)" courseTag
+    case course of
+        Right name -> name
+        Left _ -> ""
+
+findCourseFromTag :: P.Parsec String () String
```

Comment:
I recommend creating a type alias `Parser = P.Parsec String ()` to use in this file.